cirq-google/cirq_google/workflow/processor_record.py

```diff
@@ -0,0 +1,194 @@
+        return cg.engine.SimulatedLocalProcessor(
+            processor_id=self.processor_id,
+            sampler=self.get_sampler(),
+            device=self.get_device(),
+        )
+
+    def get_device(self) -> 'cirq.Device':
+        """Return a `cg.SerializableDevice` for the specified processor_id.
+
+        This method presumes the GOOGLE_CLOUD_PROJECT environment
+        variable is set to establish a connection to the cloud service.
+        """
+        return cg.get_engine_device(self.processor_id)
+
+    def get_sampler(
+        self,
+    ) -> 'cirq.Sampler':
+        """Return a local `cirq.Sampler` based on the `noise_strength` attribute.
+
+        If `self.noise_strength` is `0` return a noiseless state-vector simulator.
+        If it's set to `float('inf')` the simulator will be `cirq.ZerosSampler`.
+        Otherwise, we return a density matrix simulator with a depolarizing model with
+        `noise_strength` probability of noise.
+        """
+        if self.noise_strength == 0:
+            return cirq.Simulator()
+        if self.noise_strength == float('inf'):
+            return cirq.ZerosSampler()
+
+        return cirq.DensityMatrixSimulator(
+            noise=cirq.ConstantQubitNoiseModel(cirq.depolarize(p=self.noise_strength))
+        )
+
+    @classmethod
+    def _json_namespace_(cls) -> str:
+        return 'cirq.google'
+
+    def _json_dict_(self):
+        return cirq.dataclass_json_dict(self)
+
+    def descriptive_name(self) -> str:
+        """A pretty string name combining processor_id and noise_strength into a unique name."""
+        if self.noise_strength == 0:
+            suffix = 'simulator'
+        elif self.noise_strength == float('inf'):
+            suffix = 'zeros'
+        else:
+            suffix = f'p={self.noise_strength:.3e}'
+        return f'{self.processor_id}-{suffix}'
+
+    def __repr__(self):
+        return dataclass_repr(self, namespace='cirq_google')
+
+
+_DEVICES_BY_ID = {
+    'rainbow': cg.Sycamore23,
+    'weber': cg.Sycamore,
+}
+
+
+class SimulatedProcessorWithLocalDeviceRecord(SimulatedProcessorRecord):
+    """A serializable record mapping a processor_id and optional noise spec to a
+    completely local cg.AbstractProcessor
+
+    Args:
+        processor_id: The processor id we are emulating
+        noise_strength: To do noisy simulation, set this to a positive float. The default
+            of `0` will result in a noiseless state-vector simulation. If `float('inf')`
+            is provided the simulator will be `cirq.ZerosSampler`. Otherwise, use
+            a depolarizing model with this probability of noise.
+    """
+
+    def get_device(self) -> 'cirq.Device':
+        """Return a `cg.SerializableDevice` for the specified processor_id.
+
+        Only 'rainbow' and 'weber' are recognized processor_ids and the device information
+        may not be up-to-date, as it is completely local.
+        """
+        return _DEVICES_BY_ID[self.processor_id]
```

Comment:
Do you want to use the devices in [devices/specifications](https://github.com/quantumlib/Cirq/tree/master/cirq-google/cirq_google/devices/specifications)?  

For instance, there is this call which is very similar:
[create_noiseless_virtual_engine_from_latest_templates](https://github.com/quantumlib/Cirq/blob/master/cirq-google/cirq_google/engine/virtual_engine_factory.py#L213)

Reply:
yes I do.

I'm going to try to see what I have to add to that module so I can request the latest template by name instead of getting them in a list.

Reply:
I think I'd need something like https://github.com/quantumlib/Cirq/pull/4909 to re-use as much as possible

Reply:
merged in #4909 to use virtual_engine_factory stuff to make these local devices